### PR TITLE
Separate persistent disks

### LINKED ISSUE #743: Separate persistent data to individual disks on the vSphere Integrated Containers Appliance
Tracking for separating the MySQL DB for the registry and management portal to second vDisk and separate the log files to a third vDisk.

Use case to mitigate issues if the root disk becomes full.

Bug Ref# 1954150

### PR DESCRIPTION
**Pull request by morris-jason**
Fixes #743, consistent with storage design proposals in #954.

Also includes a hot fix for failing fast in the cache script if a curl request returns an error.
Adds the `-it` flag to building the docker container locally.

### CODE REVIEW

File: installer/build/baseimage/stage.sh [hunk 1]
```diff
@@ -320,47 +318,52 @@ done
...
-convert "$dev2" "$DATAROOT" "$IMG2NAME" "$VMDK2NAME"
+section "export images to VMDKs"
+for i in "${!IMAGES[@]}"; do
+  progress "exporting ${IMAGES[$i]}.img to ${IMAGES[$i]}.vmdk"
+  convert "${DEVS[$i]}" "${IMAGEROOTS[$i]}" "${IMAGES[$i]}.img" "${IMAGES[$i]}.vmdk"
+done
 
 task "VMDK Sizes"
-du -h bin/*
+du -h bin/*
```
Comment (andrewtchin) [inline installer/build/baseimage/stage.sh]: is this used to set the size in the ovf at the end? i think that's not happening correctly on master

File: installer/build/scripts/provision_harbor.sh [hunk 2]
```diff
@@ -38,7 +38,11 @@ for _, s in enumerate(dataMap["services"]):
         dataMap["services"][s]["restart"] = "on-failure"
   if "volumes" in dataMap["services"][s]:
     for kvol, vol in enumerate(dataMap["services"][s]["volumes"]):
-      if vol.startswith( '/data' ):
+      if vol.startswith( '/data/database' ):
```
Comment (andrewtchin) [inline installer/build/scripts/provision_harbor.sh]: can you put a comment to note that this is fixing up the compose file?

File: installer/build/scripts/systemd/engine_installer/engine_installer.service [hunk 3]
```diff
@@ -1,8 +1,8 @@
 [Unit]
 Description=Demo VCH Installer Wizard Service
 Documentation=https://github.com/vmware/vic
-After=engine_installer_startup.service data.mount
-Requires=data.mount
+After=engine_installer_startup.service .targetvic-mounts.target
```
Comment (andrewtchin) [inline installer/build/scripts/systemd/engine_installer/engine_installer.service]: is this a typo?

**Comment by andrewtchin** [comment]
reminder to ping stuart to update docs and system requirements for this

File: installer/build/baseimage/stage.sh (hunk 1 above)
Comment (morris-jason) [reply installer/build/baseimage/stage.sh]: No, it just prints out the value. ovf vmdk file reference sizes are definitely not correct in master, and haven't been for a long time.

File: installer/build/scripts/provision_harbor.sh (hunk 2 above)
Comment (morris-jason) [reply installer/build/scripts/provision_harbor.sh]: sure.

File: installer/build/scripts/systemd/engine_installer/engine_installer.service (hunk 3 above)
Comment (morris-jason) [reply installer/build/scripts/systemd/engine_installer/engine_installer.service]: yes, good catch!

**Comment by andrewtchin** [comment]
Remove this since it's not needed for upgrade anymore https://github.com/vmware/vic-product/blob/master/installer/build/cache.sh#L31

**Comment by andrewtchin** [comment]
Also if logs will be lost during the upgrade to new disks, print a warning in the upgrade script with a confirm y/n input

File: installer/build/scripts/upgrade/util.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
```
Comment (andrewtchin) [inline installer/build/scripts/upgrade/util.sh]: nice

**Pull request merged**